saleae/saleae.py

```diff
@@ -379,15 +382,20 @@ def get_active_channels(self):
 		return digital, analog
 
 	def set_active_channels(self, digital, analog):
+		digital_no = 0 if digital is None else len(digital)
+		analog_no = 0 if analog is None else len(analog)
+		if digital_no <= 0 and analog_no <= 0:
+			raise ValueError('bad active channels, has to set at least one active channel')
```

Comment:
I agree, previously the code was mainly using NotImplementedError errors in that area and I did not see the ImpossibleSettings exception.

Just for uniformity, why is ImpossibleSettings plural and not singular? Any particular reasons?